File: Data Structures & Algorithms/jump-game/submission-1.py
class Solution:
    '''
    target - 4 
    index + nums[index] >= 
    3 + 1 >= 4
    target = 3
    '''
    def canJump(self, nums: List[int]) -> bool:
        goal = len(nums) - 1
        for index in range(len(nums) - 2, -1, -1 ):
            if index + nums[index] >= goal :
                goal = index
        return goal == 0 

File: Data Structures & Algorithms/concatenation-of-array/submission-1.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        
        int n = nums.size();
        vector<int> ans(2*n);
        // for (int index = 0 ; index < 2*n ; index++){
        //     int num_index = index % nums.size() ;
        //     ans[index] = nums[num_index];
        // }

        for (int i = 0 ; i < n ; i++){
            ans[i] = nums[i];
            ans[i+n] = nums[i];
        }
        return ans;
    }
};